test/controllers/respondent_controller_test.exs

```diff
@@ -903,17 +917,48 @@ defmodule Ask.RespondentControllerTest do
       csv = response(conn, 200)
 
       [line1, line2, _] = csv |> String.split("\r\n")
-      assert line1 == "Respondent ID,Date,Modes,Smokes,Exercises,Perfect Number,Question,Disposition,Total sent SMS,Total received SMS"
+      assert line1 == "Respondent ID,Date,Modes,Sample File,Smokes,Exercises,Perfect Number,Question,Disposition,Total sent SMS,Total received SMS"
 
-      [line_2_hashed_number, _, line_2_modes, line_2_smoke, line_2_exercises, _, _, line_2_disp, _, _] = [line2] |> Stream.map(&(&1)) |> CSV.decode |> Enum.to_list |> hd
+      [line_2_hashed_number, _, line_2_modes, line_2_respondent_group, line_2_smoke, line_2_exercises, _, _, line_2_disp, _, _] = [line2] |> Stream.map(&(&1)) |> CSV.decode |> Enum.to_list |> hd
 
       assert line_2_hashed_number == respondent_1.hashed_number
       assert line_2_modes == "SMS, Phone call"
+      assert line_2_respondent_group == group_1.name
       assert line_2_smoke == "Yes"
       assert line_2_exercises == "No"
       assert line_2_disp == "Partial"
     end
 
+    test "download results csv with sample file column and two different respondent groups", %{conn: conn, user: user} do
```

Comment:
Given we had to test respondent groups in every other test, let's raise the bar a little bit in here: test multiple respondents per group sharing the value, and maybe test some whitespace/punctuation in the file name.